doc/s3ip_sysfs/s3ip_sysfs_hld.md

```diff
@@ -0,0 +1,309 @@
+
+    ret = s3ip_sysfs_watchdog_drivers_register(&drivers);
+    if (ret < 0) {
+        WDT_ERR("watchdog drivers register err, ret %d.\n", ret);
+        return ret;
+    }
+    WDT_INFO("watchdog create success.\n");
+    return 0;
+}
+
+static void __exit watchdog_dev_drv_exit(void)
+{
+    s3ip_sysfs_watchdog_drivers_unregister();
+    WDT_INFO("watchdog_exit success.\n");
+    return;
+}
+
+module_init(watchdog_dev_drv_init);
+module_exit(watchdog_dev_drv_exit);
+```
+
+##### 3.2.2 s3ip sysfs service
+Workflow of the S3ip sysfs service
+- Delete old data, such as the /sys_switch directory and subdirectories
+- Install kernel modules, including s3ip_sysfs.ko and other driver kernel modules
+- Parse the configuration file and create the /sys_switch directory
+
+The S3ip sysfs service (aka s3ip_sysfs.service) configuration file is designed to simplify the /sys_switch directory creation process.
+The configuration file contains the following metadata:
+- path: Indicates the specific path of /sys_switch
+- type: Indicates the type of the path 
+  - string: fixed string type
+  - path: real path type
+- value: indicates the real path or fixed string
+
+Sample configuration file to make /sys_switch witch S3ip sysfs Service 
+
+```
+{
+  "s3ip_syfs_paths": [
+    # The /sys_switch/fan/num path specify the number of fans on the platform. This configuration makes the path read-only and return "6" in string format.
+    {
+      "path": "/sys_switch/fan/num",
+      "type" : "string",
+      "value" : "6",
+      "description": "number of fan"
+    },
+    # The /sys_switch/syseeprom path specify the syseeprom information. This configuration makes the path linked to /sys/class/i2c_adapter/i2c-block/hwmon/hwmon1/eeprom2/2-0048, so we can reuse old driver sysfs path
+    {
+      "path":"/sys_switch/syseeprom",
+      "Type":"path",
+      "Value":"/sys/class/i2c_adapter/i2c-block/hwmon/hwmon1/eeprom2/2-0048"
+    },
+    # the /sys_switch/fan give you all fan infomations. This configuration makes the path linked to /sys/s3ip/fan, so we can use driver base on the s3ip sysfs kernel module framework
+    {
+      "path":"/sys_switch/fan ",
+      "Type":"path",
+      "value":"/sys/s3ip/fan"
+    }
+  ]
+}
+```
+### 3.3 Porting guide
+1. git clone sonic-platform-common to get the S3IP framework
+2. Verify the availability of the S3IP framework on the host computer
+ - Generate a host package, run the following command
+     - dpkg-buildpackage -rfakeroot -b -US -UC
+ - Install the package, run the following command
+     - dpkg -i s3ip-sysfs_1.0.0_amd64.deb
+ - Check the /sys_switch directory, run the following command
+     - tree -psv /sys_switch/
+3. Porting S3IP sysfs framework to platform project
+ - Create the S3IP sysfs service configuration file. 
+     - Modify /s3ip/s3ip_sysfs_service/s3ip_sysfs_conf.json to create a configuration file
+     - This file should be installed to /etc/s3ip/s3ip_sysfs_conf.json on the device

```

Comment:
Like I mentioned in the review for PIT:
https://github.com/sonic-net/SONiC/pull/1014/files#r932428427

> I am concerned that this will be yet another configuration file that platform vendors will have to generate and maintain when bringing platforms in to SONiC. Can the platform configuration be derived from already-existing platform configs?

Reply:
We plan to be two-in-one with the PDDF framework. but in two steps. The first step was to integrate the S3IP SYSFS framework into the Sonic community.
At present, there are many vendors in china that comply with S3IP framework. The S3IP SYSFS Framework is integrated into the SONiC community so that Chinese ODM vendors can more easily contribute their existing platforms to the community,  and other vendors can also have the opportunity to adapt their platforms according to S3IP specifications and expand their business opportunities in China. This is good for SONiC community ecology;